### ISSUE TITLE
[S3] An upload example with meta data

### ISSUE THREAD

**Issue by jaydeep17** [awslabs/aws-sdk-android-samples#27]
I'd really appreciate an example of uploading a file to S3 with metadata using TransferUtility.

Currently I set cache control on files while uploading

```java
ObjectMetadata omd = new ObjectMetadata();
omd.setCacheControl("public, max-age=31536000");
```

**Comment by wdane** [comment]
Currently the Transfer Utility does not support metadata.  But I will leave this open as a feature request, and we'll definitely take it into consideration!

**Comment by angelromero** [comment]
Yes please, it should be a must have feature, this question is from 15 days ago, is it now available? :)

**Comment by wdane** [comment]
Not yet, but knowing that more people want it will certainly bump it's priority!

**Comment by angelromero** [comment]
Is it possible to create a TransferObserver in wait state? with current TransferObserver constructor we need to specify an id of an existing observer, I'd like to create an observer an still be persisted by transferutility so I can be able to run the upload later.
I have a sync button that would upload all the files added to transferutility db.
Currently I see no way to do it, any clue?
Thanks!

**Comment by wdane** [comment]
@angelromero 

Please post new questions as either a new issue, on either the forums (https://forums.aws.amazon.com/forum.jspa?forumID=88&start=0), or on stack overflow (with tags amazon-web-services;android).  That way other people can easily find the answer and this thread doesn't get clogged.

But to answer this question.  When you create a new TransferObserver with the id, it does not change the state.  So if the transfer is in the wait state, it will still be in that state when you create the observer.  If you are saying you would like to create a new TransferObserver for something that has never been partially uploaded, then currently that is not possible.  The waiting state is really only applicable when you've told Transfer Utility to start a transfer, and the transfer is currently waiting on the Android Service to begin execution. (usually short time, but could be longer if you're uploading many things and it's waiting on threads in the thread pool to become available).

You can easily have a sync all button but calling resume on all transfers returned from getTransferWithTypeAndState(TransferType.UPLOAD, TransferState.PAUSED).  If you want a user to select a bunch of files prior to starting the upload, just hold onto the file paths and call upload on all the files paths once the user has hit sync all.

**Comment by wdane** [comment]
With the release of 2.2.6 of the SDK (https://aws.amazon.com/releasenotes/Android/3264977967699398).  The TransferUtility now supports adding Object Metadata to uploads requests.  I can leave this issue open as a request for a sample showcasing the new API, but it is not possible.  Look for the upload(bucketName, key, file, objectMetadata) API.  You can retrieve object metadata using the low level S3 API getObjectMetadata(bucket, key).

**Comment by jaydeep17** [comment]
Works like charm! thanks :+1:
